### PR TITLE
Add {update-verify}-bazel.sh scripts for CI

### PR DESCRIPTION
**Pull request by vincepri**
Signed-off-by: Vince Prignano <vince@vincepri.com>

**What this PR does / why we need it**:
Given that Bazel is being reintroduced in #601, this PR adds update/verify bazel script based on the work on cluster-api-provider-aws and k/k.

**Which issue(s) this PR fixes** *(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close the issue(s) when PR gets merged)*:

**Special notes for your reviewer**:

_Please confirm that if this PR changes any image versions, then that's the sole change this PR makes._

**Release note**:

```release-note

```

### CODE REVIEW

File: hack/verify-bazel.sh [hunk 1]
```diff
@@ -0,0 +1,42 @@
...
+
+_tmp_gopath="${_tmpdir}/go"
+_tmp_kuberoot="${_tmp_gopath}/src/sigs.k8s.io/cluster-api-provider-aws"
+mkdir -p "${_tmp_kuberoot}/.."
+cp -a "${KUBE_ROOT}" "${_tmp_kuberoot}/.."
+
+cd "${_tmp_kuberoot}"
+GOPATH="${_tmp_gopath}" PATH="${_tmp_gopath}/bin:${PATH}" "${KUBE_ROOT}/hack/update-bazel.sh"
+
+diff=$(diff -Naupr --exclude=".git" --exclude="bazel-*" -x '_output' "${KUBE_ROOT}" "${_tmp_kuberoot}" || true)
```
Comment (justinsb) [inline hack/verify-bazel.sh]: I think you can also do

`bazel run //:gazelle -- update -mode diff 2>/dev/null`

Bizarrely for me, I see `bazel run //:gazelle -- update` behaving differently from `bazel run //:gazelle`, even though update is supposed to be the default.  Not sure what's going on there

Comment (justinsb) [reply hack/verify-bazel.sh]: (The hope would be we could avoid the init.sh script)

Comment (justinsb) [reply hack/verify-bazel.sh]: Aha!  So when we pass `update`, it doesn't pick up the `external` attribute from the gazelle bazel rule.

So we need:  `bazel run //:gazelle -- update -mode diff -external vendored` 

(sorry to bombard you with comments!)

Comment (vincepri) [reply hack/verify-bazel.sh]: OOoooooOOo that's so much better! Thanks, TIL

**Comment by vincepri** [comment]
@justinsb  Updated and pushed

File: hack/lib/init.sh [hunk 2]
```diff
@@ -0,0 +1,132 @@
+#!/usr/bin/env bash
```
Comment (justinsb) [inline hack/lib/init.sh]: I think this is now unused?

Comment (justinsb) [inline hack/lib/init.sh]: I think this is now unused?

**Comment by justinsb** [comment]
So I'm wondering if we should remove the init.sh?  I'm sure it's fine, and I suspect we'll want it eventually, but it is a big block of bash that I find hard to understand :-)

**Comment by vincepri** [comment]
Gonna remove it, you're right I forgot. Getting more caffeine as we speak :)

**Comment by justinsb** [comment]
Cheers to that @vincepri ;-)  Thanks for doing this!

/lgtm

**Comment by vincepri** [comment]
Perfect, I'll PR against test-infra later today

**Pull request merged**